src/core/core.legend.js

```diff
@@ -437,10 +439,18 @@ module.exports = function(Chart) {
 
 					if (x >= hitBox.left && x <= hitBox.left + hitBox.width && y >= hitBox.top && y <= hitBox.top + hitBox.height) {
 						// Touching an element
-						if (opts.onClick) {
-							opts.onClick.call(me, e, me.legendItems[i]);
-						}
-						break;
+						if (e.type === 'mouseup' || e.type === 'click') {
+					        if (opts.onClick) {
+					            opts.onClick.call(me, e, me.legendItems[i]);
+					        }
+					        break;
+					    }
+					    else if(e.type === 'mousemove') {
```

Comment:
Would be better to stick to the surrounding code style and use the following format:

    if (condition) {
        // ...
    } else {
        // ...
    }

Not a big deal, but more pleasant when code formatting is consistent :)